ls1intum/Artemis
# [Feature] Detect Default Git Branch

### Checklist

### Motivation and Context


Closes #2699 

### Description

Instead of assuming `master` as the default branch we use a branch that is configured in the Artemis config file.

### Steps for Testing

1. Log in to Artemis
2. Check an old programming exercise, that was created before the deployment of this PR
3. Check a newly created programming exercise
4. Check that it still works after a server restart when this PR is and was deployed

A programming exercise works if:
- You can start it with a new student
- Continue the participation, if you started the exercise before deploying this PR
- Can use the code editor as student and editor
- Can clone and push the exercise using your IDE
- The tests are executed when a new push is made from the IDE or the online code editor

**18.06: The default branch is currently set to `main `- trying out if old exercises (they have been created with a master branch) still work properly is a good test case!**

### Test Coverage



GitService: 82% -> 85%
GitLabService: 85% -> 84%
BitBucketService: 76% -> 75%
AbstractVersionControlService: 92% -> 93%

### Screenshots
